client/components/site-verticals-suggestion-search/index.jsx

```diff
@@ -40,6 +42,46 @@ export class SiteVerticalsSuggestionSearch extends Component {
 		};
 	}
 
+	componentDidMount() {
+		this.props.initialValue && this.props.requestVerticals( this.props.initialValue, 1 );
+	}
+
+	componentDidUpdate( prevProps ) {
+		// Check if there's a direct match for any subsequent
+		// HTTP requests
+		if ( prevProps.lastUpdated !== this.props.lastUpdated ) {
+			this.updateVerticalData(
+				this.searchInResult( this.state.searchValue ),
+				this.state.searchValue
+			);
+		}
+	}
+
+	// When a user is keying through the results,
+	// only update the vertical when they select a result.
+	searchInResult = ( value = '' ) =>
+		find(
+			this.props.verticals,
+			item => item.vertical_name.toLowerCase() === value.toLowerCase() && ! isEmpty( item.preview )
+		);
+
+	updateVerticalData = ( result, value ) => {
+		// If there's not direct match
+		// return a default model
+		const verticalData = result || {
+			vertical_name: value,
+			vertical_slug: value,
+			is_user_input_vertical: true,
+		};
```

Comment:
So this part is actually tricky, because WP convention is snake_case and this is mocking the API response, which is properly using that. It does make for a bit of cognitive dissonance but I think it's best to leave it untransformed because it should signal that this is mapping to API response data.

Reply:
>  since we are using camelCase as property naming convention throughout calypso.

Yeah, I'm aware  we're doing this "camel to snake case and back again" conversion across the app to conform to our style guide, but had to leave it here for reasons @mattwiebe has mentioned.